packages/api/src/car.js

```diff
@@ -116,22 +120,23 @@ export async function handleCarUpload (request, env, ctx, car, uploadType = 'Car
 
   // Throws if CAR is invalid by our standards.
   // Returns either the sum of the block sizes in the CAR, or the cumulative size of the DAG for a dag-pb root.
-  const { size: dagSize, rootCid, structure } = await carStat(carBytes)
+  const { size: dagSize, rootCid, carCid, structure, linkIndex, blockOffsets } = await carStat(carBytes)
 
   const sourceCid = rootCid.toString()
   const contentCid = normalizeCid(sourceCid)
-  const carCid = CID.createV1(CAR_CODE, await sha256.digest(carBytes))
   const s3Key = `raw/${sourceCid}/${user._id}/${toString(carCid.multihash.bytes, 'base32')}.car`
   const r2Key = `${carCid}/${carCid}.car`
   env.sentry?.setExtras({ sourceCid, carCid: carCid.toString(), dagSize, structure, s3Key, r2Key })
 
-  await Promise.all([
+  const [,, { cid: indexCid, carCid: indexCarCid }] = await Promise.all([
     putToS3(env, s3Key, carBytes, carCid, sourceCid, structure),
     putToR2(env, r2Key, carBytes, carCid, sourceCid, structure),
-    writeSatNavIndex(env, carCid, carBytes),
+    writeSatNavIndex(env, carCid, blockOffsets),
     writeDudeWhereIndex(env, contentCid, carCid)
   ])
 
+  await writeContentClaims(env, rootCid, carCid, indexCid, indexCarCid, structure, linkIndex)
```

Comment:
could this be done in parallel as above? We should have enough concurrent sub-requests

Reply:
We need the index CID from the put to SATNAV, but I guess we could parallelise with the rest...